### ISSUE TITLE
 Can't call findElementFromElement on an existing element found with $$(...)

### ISSUE THREAD

**Issue by lacell75** [webdriverio/webdriverio#7211]
[*Environment (please complete the following information):**
 - **WebdriverIO version:** 7.9.0
 - **Mode:** WDIO
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 14.16
 - **NPM version:** 6.14.11
 - **Browser name and version:** Any
 - **Platform name and version:**  Windows 10
 - **Additional wdio packages used (if applicable):** "@wdio/cli": "^7.9.0",
    "@wdio/cucumber-framework": "^7.9.0",
    "@wdio/dot-reporter": "^7.9.0",
    "@wdio/local-runner": "^7.9.0",
    "@wdio/mocha-framework": "^7.9.0",
    "@wdio/spec-reporter": "^7.9.0",
    "@wdio/sync": "^7.9.0",
    "@wdio/allure-reporter": "^7.9.0",
     "expect": "^27.0.6",
    "expect-webdriverio": "^3.1.0",

**Describe the bug**
After upgrade to  7.9.0` (was working in `7.8.0`), I can't call an element from element found with $$(...).

**To Reproduce**
Sample feature file:

```
Feature: Find a element in table
  Scenario: Test
    Given test element if existing

```
code snippet:
```

import { Given } from '@cucumber/cucumber';

Given('test element if existing', () => {
  browser.url('https://getbootstrap.com/docs/4.1/content/tables/');
  const element = $$('tr')
    .find((el) => el.getText().includes('Mark'))
    .$('td:nth-of-type(3)');
  console.log(element.isExisting());     /// true
  console.log(element.$('.RNNXgb').isExisting());   /// An Error is displayed  instead of return `false`
});



```

**Expected behavior**
element.$('.RNNXgb').isExisting() should return false

**Log**
```
[0-0] true
[0-0] Error in "0: test element if existing
Error: Can't call findElementFromElement on element with selector "td:nth-of-type(3)" because element wasn't found
    at Object.implicitWait [as default] (C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\utils\implicitWait.js:34:19)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\utils\refetchElement.js:29:16
    at Element.elementErrorHandlerCallbackFn (C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\middlewares.js:37:37)
    at Element.$ (C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\commands\browser\$.js:83:17)
    at Element.elementErrorHandlerCallbackFn (C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\middlewares.js:23:32)
    at Element.runCommandWithHooks (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\wrapCommand.js:105:25)
    at Element.runCommandWithHooks (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\wrapCommand.js:100:24)
    at Element.wrapCommandFn (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\wrapCommand.js:67:44)
    at Element.<anonymous> (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\utils\build\shim.js:219:24)
    at Element.$ (C:\2-gitlab\end-to-end-tests-wdio\node_modules\webdriverio\build\middlewares.js:44:12)
    at World.<anonymous> (C:\2-gitlab\end-to-end-tests-wdio\src\steps\/given.js:116:23)
    at World.executeSync (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\index.js:38:22)
    at C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\index.js:69:68

```

**Comment by erwinheitzman** [comment]
Just tried to reproduce this with the following code:
```js
Given('test element if existing', () => {
    await browser.url('https://webdriver.io');

    const element = await $$('main section .container .col');
    const match = element.find(async (el) => (await el.getText()).includes('Compatible'));

    if (!match) throw Error('no match found.');
    const header = await match.$('h3');

    console.log(await header.isExisting());
    console.log(await (await header.$('.does-not-exist')).isExisting());
});
```

I was not able to reproduce the issue and it works just fine.

Not that at first it will not find the element which throws a WebDriver error after which your console.log comes in play and correctly logs `false`.

**Comment by praveendvd** [comment]
> Just tried to reproduce this with the following code:
> 
> ```js
> Given('test element if existing', () => {
>     await browser.url('https://webdriver.io');
> 
>     const element = await $$('main section .container .col');
>     const match = element.find(async (el) => (await el.getText()).includes('Compatible'));
> 
>     if (!match) throw Error('no match found.');
>     const header = await match.$('h3');
> 
>     console.log(await header.isExisting());
>     console.log(await (await header.$('.does-not-exist')).isExisting());
> });
> ```
> 
> I was not able to reproduce the issue and it works just fine.
> 
> Note that at first it will not find the element which throws a WebDriver error after which your console.log comes in play and correctly logs `false`.

he using sync it seems , could you have look in sync

**Comment by erwinheitzman** [comment]
With sync mode I am getting a different result as I am getting a `UnhandledPromiseRejectionWarning`.

@christian-bromann could you weigh in on this too?

Depending on how I structure the code I get such a rejection at different levels and I get a feeling that the test runner no longer waits properly with the latest version.

I reproduced this using a sync version of the code I shared above and used the same packages (dot + spec reporter, cucumber, sync mode and CDP protocol).

Not sure why my result in sync is different from the topic starter but I feel like he/she is onto something here.

**Comment by erwinheitzman** [comment]
@praveendvd I ran through quite a bit of tickets yesterday and should have taken sync into account. Thanks for keeping me sharp 👍

**Comment by christian-bromann** [comment]
fixed in #7215
